dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -130,18 +130,22 @@ def get_pivoted_dimension_columns_mapped_to_project(self):
                 mapping_config = self._dimension_mapping_mgr.get_by_id(
                     ref.mapping_id, version=ref.version
                 )
-                from_ids = set()
-                to_ids = set()
-                for record in mapping_config.model.records:
-                    if record.to_id is not None:
-                        from_ids.add(record.from_id)
-                        to_ids.add(record.to_id)
-
+                records = mapping_config.get_records_dataframe()
+                from_ids = get_unique_values(records, "from_id")
+                to_ids = get_unique_values(
+                    records.select("to_id").filter("to_id IS NOT NULL"), "to_id"
+                )
                 diff = from_ids.difference(columns)
                 if diff:
                     raise DSGInvalidDataset(
                         f"Dimension_mapping={mapping_config.config_id} has more from_id records than the dataset pivoted {dim_type.value} dimension: {diff}"
                     )
+                diff = from_ids.symmetric_difference(self.get_pivoted_dimension_columns())
```

Comment:
this should be `diff = columns.difference(from_ids)`, `(from_ids).difference(columns)` is already being checked above

Reply:
Right. I changed it to be a single check. We don't need both. If needed, I could add a print of the difference going in each direction.